### ISSUE TITLE
Fix all the ignored test cases[Blocked on #4991]

### ISSUE THREAD

**Issue by anandwana001** [oppia/oppia-android#2417]
**Describe the bug**
Some of the test cases are ignored due to flakyness or blocked somewhere in other pr.

**Expected behavior**
Need to fix all.

| Test file | Test issue/pr |
|---------|-----------|
| AudioFragmentTest |  #56(Landscape) | 
| ExplorationActivityTest | 2 - fake audio library (https://github.com/oppia/oppia-android/issues/2480) | 
| StateFragmentTest | 3 - #1612, 1 - #1611 | 
| ImageRegionSelectionInteractionViewTest | 3 - #1611 , 3 - test doesn't seems correct | 
| InputInteractionViewTestActivityTest | 4 - #56(Landscape) | 
| NavigationDrawerActivityTest | 1 - #1806(MyDownloads)[BLOCKED]| 
| CoroutineExecutorServiceTest | 2 - #1763 | 

| Test file | `@RunOn(TestPlatform.ESPRESSO)` |
|-----------|------------------|
| NavigationDrawerActivityTest | 2 -  Error `'is drawer closed' doesn't match the selected view.` for robolectric |
| TopicInfoFragmentTest | 7 - #2057 Text lines/Orientation change for robolectric |
| QuestionPlayerActivityTest. | 4 - #2057 |
| AudioFragmentTest | 1 - https://github.com/oppia/oppia-android/issues/2568 | 

| Test file | Need Tests to write |
|-----------|------------------|
| QuestionPlayerActivityTest | https://github.com/oppia/oppia-android/issues/503  |
|ExplorationActivityTest  | TODO(#163): Fill in remaining tests for this activity. (is this completed) |
| StateFragmentTest | TODO(#388): Add more test-cases (@khushal1707 ) ,TODO(#56): Add support for testing that previous/next button states are properly retained on config changes. |

**Comment by anandwana001** [comment]
@ShridharGoel Do you want to work on `ExplorationActivityTest` `explorationActivityTestRule.activity.isFinishing` part

**Comment by ShridharGoel** [comment]
@anandwana001 Yes, sure.

**Comment by MaskedCarrot** [comment]
@anandwana001 may I file a new issue regarding the failing robolectric test in #2499

**Comment by anandwana001** [comment]
Sure and please also mention here in the comment with the new issue link.

**Comment by MaskedCarrot** [comment]
Failing roboletric tests needs to be fixed in AudioFragmentTest #2568

**Comment by khushal1707** [comment]
@anandwana001 Is `TODO(#388): Add more test-cases` available now ?  Can I work on this?

**Comment by anandwana001** [comment]
@khushal1707 done

**Comment by khushal1707** [comment]
I'm having exams this week and I'll try to make pull request by the end of the week for TODO:(#388)
